File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left,right = 0,len(height)-1
        max_area = 0
        while left != right:
            curr_area = min(height[right],height[left])*(right-left)

            max_area = max(max_area,curr_area)
            
            if height[left]< height[right]:
                left +=1
            else:
                right -=1



        return max_area
    



















        # left = 0 
        # right = len(height)-1
        # area = 0 
        # while left <right :
        #     currentArea = min(height[left],height[right])*(right-left)
        #     area        = max(currentArea,area)

        #     if height[left] < height[right] :
        #         left += 1
        #     else: 
        #         right -= 1

        # return area

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start = 0;
        int end = height.length-1;
        int area = 0;
        int max_area = 0;
        while(start < end){
            area = (end-start)*(Math.min(height[end],height[start]));
            max_area = Math.max(max_area,area);
            if(height[start]>height[end]){
                end--;
            }
            else{
                start ++;
            }
        }
        return max_area;
    }
}